src/components/AppStyle.tsx

```diff
@@ -256,6 +261,18 @@ const CssContainer = styled.div`
  *   - children: All the children that should be rendered with the applied styling.
  */
 export default function AppStyle(props: PropsWithChildren<AppStyleProps>): ReactElement {
+  // Provide a ref for modals to use as a container. This allows them to float over
+  // other elements in the app, and lets us use the <Modal> component from Ant instead
+  // of the static modal API through Ant.App, which is less flexible.
+  const modalContainerRef = useRef<HTMLDivElement>(null);
+
+  // Force a rerender so the context provider updates with the new div ref; otherwise
+  // the value will always be null.
+  const [, setForceRerender] = useState(false);
+  useEffect(() => {
+    setForceRerender(true);
+  }, []);
```

Comment:
This mechanism to immediately trigger a second render after the initial render is a bit janky. A bit of research suggests a slightly nicer way to accomplish this might be through a [`ref` callback](https://react.dev/reference/react-dom/components/common#ref-callback), like so:

```tsx
const [modalContainer, setModalContainer] = useState(undefined);
// ...
return (<div ref={setModalContainer}>);
```

This has the added bonus of tying updates of `modalContainer` to the DOM element it actually captures, so if for some reason React decides to destroy and recreate that `div` during the lifecycle of this component, the reference will stay current.

I'm still not a huge fan of re-rendering this component, and potentially many of its children, twice (four times in dev mode!) on mount, but that might be the price of a fix here.